### PR TITLE
 oci-cli: init at 2.23.0 

### PR DESCRIPTION
**Pull request by ilian**
###### Motivation for this change
Add the [official cli tool](https://docs.oracle.com/en-us/iaas/Content/API/Concepts/cliconcepts.htm) to interact with the Oracle Cloud Infrastructure.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119412` at bb9a91bf run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>oci-cli</li>
    <li>python38Packages.oci</li>
    <li>python39Packages.oci</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/oci/default.nix:45:0:
```
   |
45 |     description = "Oracle Cloud Infrastructure Python SDK";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/oci/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    substituteInPlace setup.py \
+      --replace "configparser==4.0.2" "configparser"
+
+    # Upstream PR to update cryptography: https://github.com/oracle/oci-python-sdk/pull/322
+    substituteInPlace setup.py \
+      --replace "cryptography==3.2.1" "cryptography"
+
+    # Tracking version pinning reason of pyOpenSSL in https://github.com/oracle/oci-python-sdk/issues/339
+    substituteInPlace setup.py \
+      --replace "pyOpenSSL>=17.5.0,<=19.1.0" "pyOpenSSL"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/oci/default.nix:34]:
```suggestion
    substituteInPlace setup.py \
      --replace "configparser==4.0.2" "configparser" \
      --replace "cryptography==3.2.1" "cryptography" \
      --replace "pyOpenSSL>=17.5.0,<=19.1.0" "pyOpenSSL"
```

File: pkgs/tools/admin/oci-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    substituteInPlace setup.py \
+      --replace "cryptography==3.2.1" "cryptography"
+    substituteInPlace setup.py \
+      --replace "pyOpenSSL==19.1.0" "pyOpenSSL"
+
+    # https://github.com/oracle/oci-cli/issues/369
+    substituteInPlace setup.py \
+      --replace "PyYAML==5.3.1" "PyYAML"
+    substituteInPlace setup.py \
+      --replace "six==1.14.0" "six"
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/oci-cli/default.nix:58]:
```suggestion
    substituteInPlace setup.py \
      --replace "configparser==4.0.2" "configparser" \
      --replace "cryptography==3.2.1" "cryptography" \
      --replace "pyOpenSSL==19.1.0" "pyOpenSSL" \
      --replace "PyYAML==5.3.1" "PyYAML" \
      --replace "six==1.14.0" "six"
```

File: pkgs/development/python-modules/oci/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      --replace "cryptography==3.2.1" "cryptography" \
+      --replace "pyOpenSSL>=17.5.0,<=19.1.0" "pyOpenSSL"
+  '';
+
+  propagatedBuildInputs = [
+    certifi configparser cryptography pyopenssl dateutil pytz
+  ];
+
+  # Tests fail: https://github.com/oracle/oci-python-sdk/issues/164
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/oci/default.nix:35]:
```suggestion
  doCheck = false;
  
  pythonImportsCheck = [ "oci" ];
```

**Pull request merged**